app/scripts/lib/stores/indexeddb-store.ts

```diff
@@ -0,0 +1,111 @@
+/**
+ * Wraps an IndexedDB transaction in a promise that resolves on completion or rejects on error/abort.
+ *
+ * @param tx
+ */
+function transactionPromise(tx: IDBTransaction): Promise<void> {
+  return new Promise((resolve, reject) => {
+    tx.oncomplete = () => resolve();
+    tx.onerror = () => reject(tx.error);
+  });
+}
+
+export class IndexedDBStore {
```

Comment:
Nit: TSDoc block here would be nice. And also for `remove` and `close`. Everything else here has one already it seems.

Reply:
Remove is being named `reset` in the next PR, and JSDoc comments have been added there: https://github.com/MetaMask/metamask-extension/pull/32006/files#diff-7bd38858ecdc5af91dd4d8c0bdd5d5ecee99f726f72033c566e714256dc9e31bR13-R124